File: app/src/main/java/com/byoutline/kickmaterial/managers/AccessTokenProvider.java
package com.byoutline.kickmaterial.managers;

import com.byoutline.kickmaterial.dagger.AppScope;

import javax.annotation.Nonnull;
import javax.inject.Inject;
import javax.inject.Provider;

/**
 * Created by Sebastian Kacprzak <sebastian.kacprzak at byoutline.com> on 31.03.15.
 */
@AppScope
public class AccessTokenProvider implements Provider<String> {

    @Nonnull
    private String accessToken = "";

    @Inject
    public AccessTokenProvider() {
    }

    public void set(String accessToken) {
        if (accessToken == null) {
            accessToken = "";
        }
        this.accessToken = accessToken;
    }

    @Override
    public String get() {
        return accessToken;
    }
}

File: app/src/main/java/com/byoutline/kickmaterial/managers/AccessTokenProvider.kt
package com.byoutline.kickmaterial.managers

import com.byoutline.kickmaterial.dagger.AppScope
import javax.inject.Inject
import javax.inject.Provider

/**
 * Created by Sebastian Kacprzak <sebastian.kacprzak at byoutline.com> on 31.03.15.
 */
@AppScope
class AccessTokenProvider @Inject
constructor() : Provider<String> {

    private var accessToken = ""

    fun set(accessToken: String?) {
        this.accessToken = accessToken ?: ""
    }

    override fun get(): String {
        return accessToken
    }
}
